js_modules/dagit/packages/core/src/ui/Filters/StaticSetFilter.tsx

```diff
@@ -0,0 +1,192 @@
+    getStringValue,
+  }: {
+    name: string;
+    icon: IconName;
+    renderLabel: (props: {value: TValue; isActive: boolean}) => JSX.Element;
+    getStringValue: (value: TValue) => string;
+    allValues: SetFilterValue<TValue>[];
+    initialState?: Set<TValue> | TValue[];
+  }) {
+    super(name, icon, new Set(initialState || []));
+    this.allValues = allValues;
+    this.renderLabel = renderLabel;
+    this.getStringValue = getStringValue;
+  }
+  renderActiveFilterState() {
+    return (
+      <SetFilterActiveState
+        name={this.name}
+        state={this.getState()}
+        getStringValue={this.getStringValue}
+        renderLabel={this.renderLabel}
+        onRemove={() => {
+          this.setState(new Set());
+        }}
+        icon={this.icon}
+      />
+    );
+  }
+
+  isActive(): boolean {
+    return this.getState().size > 0;
+  }
+
+  getResults(query: string): {label: JSX.Element; value: TValue}[] {
+    if (query === '') {
+      return this.allValues.map(({value}) => ({
+        label: <SetFilterLabel value={value} renderLabel={this.renderLabel} filter={this} />,
+        value,
+      }));
+    }
+    return this.allValues
+      .filter(({match}) => match.some((value) => value.toLowerCase().includes(query.toLowerCase())))
+      .map(({value}) => ({
+        label: <SetFilterLabel value={value} renderLabel={this.renderLabel} filter={this} />,
+        value,
+      }));
+  }
+
+  onSelect(value: TValue, _: () => void) {
+    if (this.getState().has(value)) {
+      const nextState = new Set(this.getState());
+      nextState.delete(value);
+      this.setState(nextState);
+    } else {
+      this.setState(new Set(this.getState()).add(value));
+    }
+    return null;
+  }
+}
+
+function SetFilterActiveState({
+  name,
+  state,
+  icon,
+  getStringValue,
+  onRemove,
+  renderLabel,
+}: {
+  name: string;
+  icon: IconName;
+  state: Set<any>;
+  getStringValue: (value: any) => string;
+  onRemove: () => void;
+  renderLabel: (value: any) => JSX.Element;
+}) {
+  const arr = React.useMemo(() => Array.from(state), [state]);
+  const label = React.useMemo(() => {
+    if (arr.length === 0) {
+      return null;
+    } else if (arr.length <= 3) {
+      return (
+        <>
+          is&nbsp;{arr.length === 1 ? '' : <>any of&nbsp;</>}
+          {arr.map((value, index) => (
+            <>
+              <FilterTagHighlightedText>{getStringValue(value)}</FilterTagHighlightedText>
+              {index < arr.length - 1 ? <>,&nbsp;</> : ''}
+            </>
+          ))}
+        </>
+      );
+    } else if (arr.length > 3) {
+      return (
+        <div>
+          is any of{' '}
+          <Popover
+            interactionKind="hover"
+            position="bottom"
+            content={
+              <Box
+                padding={{vertical: 8, horizontal: 12}}
+                flex={{direction: 'column', gap: 4}}
+                style={{maxHeight: '300px', overflow: 'auto'}}
+              >
```

Comment:
Does this box need a max width? Curious what this looks like with super long strings.